packages/@aws-cdk/pipelines/test/builds.test.ts

```diff
@@ -25,562 +26,602 @@ afterEach(() => {
+          subdirectory: 'subdir',
+        }),
+      });
 
-  // THEN
-  expect(pipelineStack).toHaveResourceLike('AWS::CodeBuild::Project', {
-    Environment: {
-      Image: 'aws/codebuild/standard:5.0',
-    },
-    Source: {
-      BuildSpec: encodedJson(deepObjectLike({
-        phases: {
-          pre_build: {
-            commands: arrayWith('cd subdir'),
-          },
+      // THEN
+      expect(pipelineStack).toHaveResourceLike('AWS::CodeBuild::Project', {
+        Environment: {
+          Image: 'aws/codebuild/standard:5.0',
         },
-        artifacts: {
-          'base-directory': 'subdir/cdk.out',
+        Source: {
+          BuildSpec: encodedJson(deepObjectLike({
+            phases: {
+              pre_build: {
+                commands: arrayWith('cd subdir'),
+              },
+            },
+            artifacts: {
+              'base-directory': 'subdir/cdk.out',
+            },
+          })),
         },
-      })),
-    },
+      });
+    });
   });
 });
 
-test.each([['npm'], ['yarn']])('%s build sets UNSAFE_PERM=true', (npmYarn) => {
-  // WHEN
-  new TestGitHubNpmPipeline(pipelineStack, 'Cdk', {
-    sourceArtifact,
-    cloudAssemblyArtifact,
-    synthAction: npmYarnBuild(npmYarn)({
-      sourceArtifact,
-      cloudAssemblyArtifact,
-    }),
-  });
-
-  // THEN
-  expect(pipelineStack).toHaveResourceLike('AWS::CodeBuild::Project', {
-    Environment: {
-      EnvironmentVariables: [
-        {
-          Name: 'NPM_CONFIG_UNSAFE_PERM',
-          Type: 'PLAINTEXT',
-          Value: 'true',
+behavior('%s build sets UNSAFE_PERM=true', (suite) => {
+  ['npm', 'yarn'].forEach(npmYarn => {
+    suite.legacy(() => {
+      // WHEN
+      new TestGitHubNpmPipeline(pipelineStack, 'Cdk', {
+        sourceArtifact,
+        cloudAssemblyArtifact,
+        synthAction: npmYarnBuild(npmYarn)({
+          sourceArtifact,
+          cloudAssemblyArtifact,
+        }),
+      });
+
+      // THEN
+      expect(pipelineStack).toHaveResourceLike('AWS::CodeBuild::Project', {
+        Environment: {
+          EnvironmentVariables: [
+            {
+              Name: 'NPM_CONFIG_UNSAFE_PERM',
+              Type: 'PLAINTEXT',
+              Value: 'true',
+            },
+          ],
         },
-      ],
-    },
+      });
+    });
   });
 });
 
-test.each([['npm'], ['yarn']])('%s assumes no build step by default', (npmYarn) => {
-  // WHEN
-  new TestGitHubNpmPipeline(pipelineStack, 'Cdk', {
-    sourceArtifact,
-    cloudAssemblyArtifact,
-    synthAction: npmYarnBuild(npmYarn)({ sourceArtifact, cloudAssemblyArtifact }),
-  });
-
-  // THEN
-  expect(pipelineStack).toHaveResourceLike('AWS::CodeBuild::Project', {
-    Environment: {
-      Image: 'aws/codebuild/standard:5.0',
-    },
-    Source: {
-      BuildSpec: encodedJson(deepObjectLike({
-        phases: {
-          build: {
-            commands: ['npx cdk synth'],
-          },
+behavior('%s assumes no build step by default', (suite) => {
```

Comment:
The output (as seen from IntelliJ) is not great, but it's not terrible either:

![Screenshot 2021-06-22 at 15 02 48](https://user-images.githubusercontent.com/288203/122938724-ea8c7600-d36a-11eb-9d0d-bfc79a829ae3.png)

If we have some time later, we can polish this.